docs/expose.md

```diff
@@ -12,8 +12,58 @@ and [Group Replication](https://dev.mysql.com/doc/refman/8.0/en/group-replicatio
 
 ## Asynchronous Replication
 
+### Exposing cluster with HAProxy
+
+Percona Operator for MySQL provides load balancing and proxy service with
+[HAProxy](https://haproxy.org) (enabled by default).
+
+![image](assets/images/exposure-haproxy.svg)
+
+You can control whether to use it or not by enabling or disabling it via the
+`haproxy.enabled` option in the `deploy/cr.yaml` configuration file.
+
+The following example turns on the asynchronous replication and enables HAProxy:
+
+```yaml
+mysql:
+  clusterType: async
+  ...
+  haproxy: 
+   enabled: true
+   size: 3
+   image: perconalab/percona-xtradb-cluster-operator:{{ release }}-haproxy
+```
+
+The resulting HAPproxy setup will contain the `cluster1-haproxy` service
+listening on ports 3306 (MySQL) and 3309 (the [proxy protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/)
+useful for operations such as asynchronous calls).
+This service is pointing to the MySQL cluster member number zero
+(`cluster1-mysql-0`) by default when this member is available. If a zero
+member is not available, members are selected in descending order of their
+numbers (e.g. `cluster1-mysql-2`, then `cluster1-mysql-1`, etc.). This
+service can be used for both read and write load, or it can also be used
+just for write load (single writer mode) in setups with split write and read
+loads.
+
+!!! note
+
+    For obvious reasons the Operator will not allow the simultaneous
+    enabling of both HAProxy and Group Replication.
```

Comment:
Let's not call it obvious reasons and explain.

In case of group replication we use MySQL router as proxy and currently doesn't support HAProxy with GR. We may add support for it in the future.